### PR TITLE
v0.5.0 release notes (and v0.4.{1,2,3})

### PR DESCRIPTION
**Pull request by chappjc**
This adds the release notes for the major v0.5 release.

This also adds the release notes for the three patch releases to 0.4.

### CODE REVIEW

File: docs/release-notes/release-notes-0.5.0.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+- Bitcoin swap acceleration using CPFP transactions to increase the swap fee rate in the event of a network fee spike. This has not been in demand much recently, but we have encountered rapid increases in Bitcoin fee rates in the past where this is valuable. (<https://github.com/decred/dcrdex/pull/1555>)
+
+- Fiat currency value display. When enabled by the user, current USD exchange rates for all assets will be used to show equivalent fiat values on the UI. (<https://github.com/decred/dcrdex/pull/1600>)
+
+- QR code display for wallet receiving addresses. (<https://github.com/decred/dcrdex/pull/1483>)
+
+- There is now a "copy" button next to the wallet's receiving address. (<https://github.com/decred/dcrdex/pull/1639>, <https://github.com/decred/dcrdex/pull/1675>)
+
+- Rescan and recover functionality for the native BTC SPV wallet. You may also now reconfigure the native BTC wallet’s “birthday” to change the scope of the rescan. (<https://github.com/decred/dcrdex/pull/1507>, <https://github.com/decred/dcrdex/pull/1439>)
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.5.0.md:67]: This was in 0.4.1

File: docs/release-notes/release-notes-0.5.0.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
+# DCRDEX v0.5.0
+
+Aug 23, 2022
+
+For a high level introduction to DCRDEX, please read the [initial release's notes](https://github.com/decred/dcrdex/blob/master/docs/release-notes/release-notes-0.1.0.md).
+
+Binaries are attached at the bottom of this release.  Just download the one for your system, extract it, and run the **dexc** executable in a terminal / command prompt.  It will start up and display a link that you should open in the web browser of your choice.  Keep the terminal with dexc opened until you are done and have signed out from the browser window.  Hit CTRL+C in the terminal to shut it down.
+
+Windows users must use v0.5.1 or newer, which contains an important fix for the user interface.
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.5.0.md:9]: Need v0.5.1 and v0.5.2 release notes too.

File: docs/release-notes/release-notes-0.5.0.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+- Bitcoin “descriptor” wallets are now supported. This only applies to users of the Bitcoin Core software, not the native BTC wallet, who also choose to create this type of wallet that is becoming the default in future versions of Bitcoin Core. Support for Bitcoin Core v23, dropped support for v0.19. (<https://github.com/decred/dcrdex/pull/1659>, <https://github.com/decred/dcrdex/pull/1735>)
+
+- A DEX server's host name and TLS certificates may now be updated. (<https://github.com/decred/dcrdex/pull/1605>, <https://github.com/decred/dcrdex/pull/1602>)
+
+- When a wallet regains peers after having lost connectivity, a notification is displayed. (<https://github.com/decred/dcrdex/pull/1701>, <https://github.com/decred/dcrdex/pull/1719>)
+
+- Reconfiguring BTC wallets is now less disruptive as it only requires an internal restart of the wallet and chain service when absolutely required. (<https://github.com/decred/dcrdex/pull/1686>)
+
+- The user interface files are now embedded in the "dexc" executable.  To use the DEX client, only the one file is required. Any "site" folders left from previous versions may be deleted. (<https://github.com/decred/dcrdex/pull/1710>)
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.5.0.md:79]: Should call this out in "Upgrading" so they know nothing is missing, and that they can delete the old site folder.

File: docs/release-notes/release-notes-0.5.0.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+- The optional `dexcctl` command line control application now has a `version` command to ensure compatibility with dexc. (<https://github.com/decred/dcrdex/pull/1645>)
+
+NOTE: While Ethereum trading is not in production yet, the native ETH wallet can be used if built from source with the `-tags lgpl` switch.
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.5.0.md:83]: Will remove even this now that the light client is broken by The Merge.

File: docs/release-notes/release-notes-0.5.0.md [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+- The server will check that a node's transaction indexes are enabled, when possible. (<https://github.com/decred/dcrdex/pull/1549>)
+
+- Fee rate estimation now has a median-based backup estimation method when the nodes fail to return a rate. (<https://github.com/decred/dcrdex/pull/1597>, <https://github.com/decred/dcrdex/pull/1655>)
+
+- The transaction search code is now smarter and blocks other routines less. (<https://github.com/decred/dcrdex/pull/1623>)
+
+- Confirmation checks are now throttled for fast blocks. (<https://github.com/decred/dcrdex/pull/1629>)
+
+- The server now begins signing untruncated messages, and ceases to recognize signatures created on truncated messages. (<https://github.com/decred/dcrdex/pull/1530>) This follows <https://github.com/decred/dcrdex/pull/1528> in v0.4.2 in which both client and server began recognizing signatures of the full messages, and <https://github.com/decred/dcrdex/pull/1529> from v0.4.3 in which clients began signing untruncated messages. The final change will be in v0.5.1 (<https://github.com/decred/dcrdex/pull/1526>), in which will cease to recognize the old signatures from servers that have not updated to v0.5.0.
```
Comment (chappjc) [inline docs/release-notes/release-notes-0.5.0.md:98]: In Upgrading, will need to describe this minimum requirement of v0.4.2 clients for v0.5+ servers.

**Review by JoeGruffins — APPROVED** [review]
Very good notes.

**Pull request merged**